### PR TITLE
feat: `general-reddit-ads` patch

### PR DESCRIPTION
**Pull request by ponces**

### CODE REVIEW

**Review by TheJeterLP — APPROVED** [review]
Looks good to me

**Comment by oSumAtrIX** [comment]
Iirc Reddit also has ads in the comments section. I assume this is not included?

File: src/main/kotlin/app/revanced/patches/reddit/ad/general/patch/GeneralAdsPatch.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patches.reddit.ad.general.annotations.GeneralAdsCompatibility
+import org.jf.dexlib2.Opcode
+import org.jf.dexlib2.builder.instruction.BuilderInstruction21c
+import org.jf.dexlib2.iface.instruction.ReferenceInstruction
+import org.jf.dexlib2.iface.reference.StringReference
+import org.jf.dexlib2.immutable.reference.ImmutableStringReference
+
+@Patch
+@Name("general-ads")
```
Comment (oSumAtrIX) [inline src/main/kotlin/app/revanced/patches/reddit/ad/general/patch/GeneralAdsPatch.kt:20]: Conflicts with another patch:
```suggestion
@Name("general-reddit-ads")
```

**Review by oSumAtrIX — CHANGES_REQUESTED** [review]

**Comment by ponces** [comment]
> Iirc Reddit also has ads in the comments section. I assume this is not included?

Yeah, that should be another topic as I think it will be a different patch and implementation...

**Comment by oSumAtrIX** [comment]
@ponces Someone reported [this error logs](https://drive.google.com/file/d/1qFUrAP_1HnJYwhDQXf638_tSgWD8ezaq/view). Do you maybe have an idea?

**Comment by TheJeterLP** [comment]
> @ponces Someone reported [this error logs](https://drive.google.com/file/d/1qFUrAP_1HnJYwhDQXf638_tSgWD8ezaq/view). Do you maybe have an idea?

Integrations shouldn't get merged in reddit btw

**Comment by oSumAtrIX** [comment]
> > @ponces Someone reported [this error logs](https://drive.google.com/file/d/1qFUrAP_1HnJYwhDQXf638_tSgWD8ezaq/view). Do you maybe have an idea?
> 
> Integrations shouldn't get merged in reddit btw

Yeah, the docs don't say it. Currently, the integrations are very separated, but you know our plans.

**Comment by Lyceris-chan** [comment]
Using the -r argument to disable resource patching seems to fix the issue, and according to my tester they don't have any ads on the main page, they did have ads on the hamburger menu idk if they had ads anywhere else.

**Comment by oSumAtrIX** [comment]
Yes, this is an error with the resource patcher.
The description of the introduced patch says `Removes general ads from the Reddit front page and subreddits`.

**Comment by ponces** [comment]
> Integrations shouldn't get merged in reddit btw

Uh... I didn't test with integrations included... Should this be considered?

> Using the -r argument to disable resource patching seems to fix the issue, and according to my tester they don't have any ads on the main page, they did have ads on the hamburger menu idk if they had ads anywhere else.

Disabling resource patching should not be needed, at least I didn't during my tests.

**Comment by oSumAtrIX** [comment]
> Uh... I didn't test with integrations included... Should this be considered?

No, your patch does not depend on the integrations.

**Pull request merged**